CONTRIBUTING.md

```diff
@@ -1,29 +1,157 @@
+## Open Development
+All work related to the application takes place on Github itself. We use [Issues](https://github.com/coderplex/coderplex/issues) to track bugs, discuss ideas and to engage open source contributors. [Projects](https://github.com/coderplex/coderplex/projects) are used to keep track of everything and is our project management tool. We maintain [Wiki](https://github.com/coderplex/coderplex/wiki) for structuring our long term thoughts. Both core team members and contributors sends a pull request which goes through the same review process. Whole process is as transparent as it can be and we strive to keep it that way.
 
-## Running this project locally
+## Development Workflow
 
-1. Make sure you have [NodeJS](https://nodejs.org/) and [Yarn](https://yarnpkg.com/en/docs/install) installed.
-    > Make sure you install node version 8 or above and check node version by running `node -v`
-1. Then clone this repo
+We welcome pull requests from beginners and seasoned javaScript developers alike!
+
+### Work on Issues
+1. Find an issue that needs assistance by searching for the [open issues](https://github.com/coderplex/coderplex/labels/help-wanted).
+2. If you decide to fix an issue, please be sure to check the comment thread in case somebody is already working on a fix. If nobody is working on it at the moment, please leave a comment stating that you intend to work on it so other people don’t accidentally duplicate your effort.
+3. If somebody claims an issue but doesn’t follow up for more than a weeks, it’s fine to take over it but you should still leave a comment.
+
+
+### Proposing a Change
+1. Open a new issue if you would like report a bug or suggest improvements.
+2. Please wait for core team members to comment on the thread. This lets us reach an agreement on your proposal before you put significant effort into it.
+
+### Prerequisites
+1. [NodeJS](https://nodejs.org/)
+    <br/>
+    Minimum version v8.0.0+
     ```bash
-    git clone https://github.com/coderplex/coderplex.git
+    To check node version
+    $ node -v
     ```
-1. `cd coderplex`
-1. `npm install`
-1. `npm run dev` to start local development server
-1. App opens at `localhost:3000`
+    Any lower version than mentioned above may results in this [error](https://github.com/coderplex/coderplex/issues/100).
+    > If you face problem updating your node then you might need a Node version manager tool. [Follow here]()
+
+2. [Yarn](https://yarnpkg.com)
+    <br/>
+    Minimum version v1.2.0+
+    <br/>
+    Installing instructions are at [official docs](https://yarnpkg.com/en/docs/install#windows-tab).
+    Use yarn over npm
+    > Our team's official policy (for now) is: We only use [Yarn](https://yarnpkg.com/en/docs/install) as our official Node package manager, and so we request you to use Yarn instead of npm and commit `yarn.lock` file.
+
+3. [Git](https://git-scm.com/download/linux)
+    <br/>
+    Familiarity with git is mandatory.
+
+### Running
```

Comment:
I think this should be labeled as 
> ### Running Locally

And I feel it should go under **Sending a Pull Request** section